nixos/modules/services/databases/mysql.nix

```diff
@@ -249,6 +249,7 @@ in
 
         after = [ "network.target" ];
         wantedBy = [ "multi-user.target" ];
+        restartTriggers = [ config.environment.etc."my.cnf".source ];
```

Comment:
Is the variable working as you would expect? `config.environment.etc."my.cnf".source` points into /nix/store to a static file. It may work when you set `restartIfChanged` for te service.

I'd guess that `/etc/my.cnf` is the correct thing to put here, however i have not yet debugged restartTriggers.

Reply:
Yes, I checked

Reply:
Used this example https://github.com/NixOS/nixpkgs/blob/master/nixos/modules/services/networking/ssh/sshd.nix#L356

Reply:
Ah okay, `X-Restart-Triggers` is actually NixOS specific and a random commit in nixpkgs actually contains the documentation: 
```
The triggers are just arbitrary strings that are included in the unit
under X-Restart-Triggers.  The idea is that if they change between
reconfigurations, switch-to-configuration will restart the unit
because its store path changed.  This is mostly useful for services
that implicitly depend on generated files in /etc.  Thus you can say

  restartTriggers = [ confFile ];

where ‘confFile’ is the derivation that generated the /etc file in
question.
```
https://github.com/NixOS/nixpkgs/commit/b606165bd9d4c8a1f82160f67bd541f064ed1ddc